Frontend/ui-driver/src/Flow.purs

```diff
@@ -1851,17 +1851,17 @@ homeScreenFlow = do
       void $ lift $ lift $ loaderText (getString PLEASE_WAIT) if status == Online then (getString SETTING_YOU_ONLINE) else if status == Silent then (getString SETTING_YOU_SILENT) else (getString SETTING_YOU_OFFLINE)
       void $ lift $ lift $ toggleLoader true
       let label = if status == Online then "ny_driver_online_mode" else if status == Silent then "ny_driver_silent_mode" else "ny_driver_offline_mode"
-      let currentTime = (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")
+          currentTime = (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")
+          isGotoEnabled = state.data.driverGotoState.gotoEnabledForMerchant && state.data.config.gotoConfig.enableGoto
       liftFlowBT $ logEventWithParams logField_ label "Timestamp" currentTime
       changeDriverStatus status
       gs <- getState
+      when (isGotoEnabled && (status == Online || status == Silent)) $ do
+        getDriverInfoApiResp <- lift $ lift $ Remote.getDriverInfoApi (GetDriverInfoReq{})
```

Comment:
use `getDriverInfoDataFromCache` and send a flag to update the local and send the result.

Reply:
`updateDriverDataToStates` this all handle inside the same function. if we are updating the cache the update the data also.

Reply:
I think we shouldn't do this change in this PR. CC @witcher-shailesh